polkadex-primitives/src/types.rs

```diff
@@ -0,0 +1,145 @@
+use codec::{Decode, Encode, Error};
+#[cfg(feature = "sgx")]
+use sgx_tstd::vec;
+
+#[cfg(feature = "sgx")]
+use sgx_tstd::vec::Vec;
+
+#[derive(Debug, Clone, Encode, Decode, PartialEq)]
+pub enum OrderType {
+    LIMIT,
+    MARKET,
+}
+
+#[derive(Debug, Clone, Encode, Decode, PartialEq)]
+pub enum OrderSide {
+    BID,
+    ASK,
+}
+
+#[derive(Debug, Clone, Encode, Decode)]
+pub enum OrderState {
+    UNFILLED,
+    PARTIAL,
+    FILLED,
+    CANCELLED,
+}
+
+// Create Order
+#[derive(Debug, Clone, Encode, Decode, PartialEq)]
+pub struct Order {
+    pub user_uid: Vec<u8>,
```

Comment:
Ideally it is String since that's how Openware defines it, but parity's codec is not working with sgx_tstd's String, I converted to vec<u8>, Let me know if you have a better solution

Reply:
For the time being, can you define a  pub type user_uid = Vec<u8> and for the others as well? We could then easily change the type when necessary.

It will become clear what types should be used once we work with them I guess.

Reply:
Resolved